release/nightly_tests/placement_group_tests/placement_group_performance_test.py

```diff
@@ -0,0 +1,158 @@
+import argparse
+import json
+import os
+
+import ray
+from ray._private.ray_microbenchmark_helpers import timeit
+
+RESOURCES_VALUE = 0.01
+
+
+def test_placement_group_perf(num_pgs, num_bundles, num_pending_pgs):
+    # Run the placement group performance benchmark given arguments.
+    ray.cluster_resources()["custom"] >= (
+        RESOURCES_VALUE * num_pgs * num_bundles)
+
+    def placement_group_create(num_pgs):
+        pgs = [
+            ray.util.placement_group(
+                bundles=[{
+                    "custom": 0.001
+                } for _ in range(num_bundles)],
+                strategy="SPREAD") for _ in range(num_pgs)
+        ]
+        [pg.wait(timeout_seconds=30) for pg in pgs]
+        for pg in pgs:
+            ray.util.remove_placement_group(pg)
+
+    print(f"Num pending pgs: {num_pending_pgs}, "
+          f"Num pgs: {num_pgs}, "
+          f"Num bundles {num_bundles}")
+
+    # Get the throughput.
+    throughput = timeit("placement group create per second",
+                        lambda: placement_group_create(num_pgs), num_pgs)
+
+    # Get fine-grained scheduling stats.
+    latencies = []
+    e2e_latencies = []
+    scheduling_attempts = []
+    for entry in ray.util.placement_group_table().values():
+        latency = entry["stats"]["scheduling_latency_ms"]
+        e2e_latency = entry["stats"]["end_to_end_creation_latency_ms"]
+        scheduling_attempt = entry["stats"]["scheduling_attempt"]
+        latencies.append(latency)
+        e2e_latencies.append(e2e_latency)
+        scheduling_attempts.append(scheduling_attempt)
+    latencies = sorted(latencies)
+    e2e_latencies = sorted(e2e_latencies)
+    scheduling_attempts = sorted(scheduling_attempts)
+
+    # Pure scheduling latency without queuing time.
+    print("P50 scheduling latency ms: "
```

Comment:
Should we consider exporting it through stats in cpp?

Reply:
They are already all available in cpp stats as histogram!

Reply:
For tests, I used this way just because it is easier than querying prometheus (and these stats are already available in ray.util.placement_group_table for ease of discovery)